src/data_loader.py

```diff
@@ -0,0 +1,160 @@
+import os
+import PIL
+import torch
+import random
+import collections
+import numpy as np
+import pandas as pd
+
+
+
+class ImageSegmentaionDataset():
+    """Image Segmentation dataset"""
+    
+    def __init__(self, image_dict, loader_type, replacement=True, transform=None):
+        """
+        Args:
+        image_dict (dict): a dict of the following format:
+                        {"ships": {"imageId": "segmentation pixels"}， “w/ships”: ["imageId"]}
+        loader_type: (string)"classification" or "segmentation"
+        replacement: (bool) sample with/without replacement
+        transform (callable, optional): Optional transform to be applied on a sample.
+        """
+        self.image_dict = image_dict
+        self.loader_type = loader_type
+        self.replacement = replacement
+        # images_list is the name 
+        self.images_list = list(image_dict["ships"].keys()) + image_dict["w/ships"]
```

Comment:
Same comment as above

Reply:
I changed "images_list" to be "image_filenames".